### DISCUSSION TITLE
I have alert pop up. Can I accept_alert in CDP Mode ?

### DISCUSSION THREAD

**Discussion by morndej** [seleniumbase/SeleniumBase#3303] [Q&A] [answered]
is the way to use CDP mode to accept alert

**Comment by mdmintz** [comment, marked as answer]
See [examples/cdp_mode/raw_handle_alerts.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/raw_handle_alerts.py)

**Reply by morndej** [reply]
Thank you very much for your response and kind assistance. I will give it a try.

**Comment by LiPingYen** [comment]
Hi @mdmintz,

First of all, thank you for your excellent work on SeleniumBase — it’s an amazing framework that continues to evolve impressively!
I’d like to propose a feature enhancement related to alert handling in CDP mode.

Currently, functions like `is_alert_present()`,  `accept_alert()`, and `dismiss_alert()` appear to require using `sb.reconnect()` to reattach the WebDriver before interacting with browser alerts.
However, this approach has a few drawbacks:

- Reconnecting the WebDriver may increase the risk of detection in certain environments.

- It can occasionally cause minor interruptions or abnormal WebDriver behavior.

- The test flow becomes less smooth, especially in continuous testing scenarios.

If SeleniumBase could natively support handling alerts directly in CDP mode (e.g., through the Chrome DevTools Protocol), it would:

- Eliminate the need for a WebDriver reconnect.

- Make the browser interaction process more seamless and natural.

- Reduce anomalies and improve stability during alert-related operations.

This improvement would significantly enhance SeleniumBase’s reliability and stealth capabilities in CDP-based automation.
Thank you for considering this feature request!

**Reply by mdmintz** [reply]
See the updated version of https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/raw_handle_alerts.py

**Reply by LiPingYen** [reply]
Hi @mdmintz,

Thanks so much for your quick and detailed reply — really appreciate your time and explanation! 🙏

I have a follow-up question:
Does `is_alert_present()` currently support CDP mode?

I’m wondering if there’s a way to check for alerts directly via CDP, such as using something like `sb.cdp.is_alert_present()` to detect whether an alert is currently open, without needing to reconnect the WebDriver.

**Reply by mdmintz** [reply]
There's no direct CDP call for that.
